src/hotspot/share/memory/iterator.hpp

```diff
@@ -256,12 +256,9 @@ class NMethodToOopClosure : public NMethodClosure {
 };
 
 class MarkingNMethodClosure : public NMethodToOopClosure {
-  bool _keepalive_nmethods;
-
  public:
-  MarkingNMethodClosure(OopClosure* cl, bool fix_relocations, bool keepalive_nmethods) :
-      NMethodToOopClosure(cl, fix_relocations),
-      _keepalive_nmethods(keepalive_nmethods) {}
+  MarkingNMethodClosure(OopClosure* cl) :
+      NMethodToOopClosure(cl, false /* fix_relocations */) {}
```

Comment:
Would it be appropriate to use `!NMethodToOopClosure::FixRelocations` here instead?